drafts/records.md

```diff
@@ -0,0 +1,398 @@
+    public float $totalSeconds {
+        get => $this->milliseconds / 1000,
+    }
+    
+    public float $totalMinutes {
+        get => $this->totalSeconds / 60,
+    }
+    /* ... */
+}
+
+$time1 = Time(1000);
+$time2 = Time(5000);
+
+echo $time1 < $time2; // Outputs: true
+```
+
+### Reflection
+
+Records in PHP will be fully supported by the reflection API,
+providing access to their properties and methods just like regular classes.
+However, immutability and special instantiation rules will be enforced.
+
+#### ReflectionClass support
+
+It can be used to inspect records, their properties, and methods. Any attempt to modify record properties
+via reflection will throw an exception, maintaining immutability. Attempting to create a new instance via
+`ReflectionClass` will cause a `ReflectionException` to be thrown.
+
+``` php
+$point = Point(3, 4);
+$reflection = new \ReflectionClass($point);
+
+foreach ($reflection->getProperties() as $property) {
+    echo $property->getName() . ': ' . $property->getValue($point) . PHP_EOL;
+}
+```
+
+#### Immutability enforcement
+
+Attempts to modify record properties via reflection will throw an exception.
+
+``` php
+try {
+    $property = $reflection->getProperty('x');
+    $property->setValue($point, 10); // This will throw an exception
+} catch (\ReflectionException $e) {
+    echo 'Exception: ' . $e->getMessage() . PHP_EOL; // "Cannot modify a record property"
+}
+```
+
+#### ReflectionFunction for implicit constructor
+
+Using `ReflectionFunction` on a record will reflect the constructor.
+
+``` php
+$constructor = new \ReflectionFunction('Geometry\Point');
+echo 'Constructor Parameters: ';
+foreach ($constructor->getParameters() as $param) {
+    echo $param->getName() . ' ';
+}
+```
+
+#### New functions and methods
+
+- Calling `is_object($record)` will return `true`.
+- A new function, `is_record($record)`, will return `true` for records, and `false` otherwise
+- Calling `get_class($record)` will return the record name
+
+### var_dump
+
+Calling `var_dump` will look much like it does for objects, but instead of `object` it will say `record`.
+
+```txt
+record(Point)#1 (2) {
+  ["x"]=>
+  int(1)
+  ["y"]=>
+  int(2)
+}
+```
+
+### Considerations for implementations
+
+A `record` cannot be named after an existing `record`, `class` or `function`. This is because defining a `record`
+creates both a `class` and a `function` with the same name.
+
+### Autoloading
+
+As invoking a record value by its name looks remarkably similar to calling a function,
+and PHP has no function autoloader, autoloading will not be supported in this implementation.
+If function autoloading were to be implemented in the future, an autoloader could locate the `record` and autoload it.
+The author of this RFC strongly encourages someone to put forward a function autoloading RFC if autoloading is desired for records.
+
+## Backward Incompatible Changes
+
+No backward incompatible changes.
```

Comment:
Might want to mention how BC changes to the keyword `record` will be bypassed as they were with `enum`.